packages/perseus/src/widgets/interactive-graphs/grid.tsx

```diff
@@ -0,0 +1,29 @@
+import {Coordinates} from "mafs";
+import * as React from "react";
+
+import type {PerseusImageBackground} from "../../perseus-types";
+import type {SizeClass} from "../../util/sizing-utils";
+
+const renderLineLabel = (n: number, [min, max]: [number, number]) =>
+    n !== -1 && n !== min && n !== max;
+
+export const Grid = (props: {
+    step: number[];
+    gridStep: number[];
+    range: [[number, number], [number, number]];
+    backgroundImage?: PerseusImageBackground;
+    containerSizeClass: SizeClass;
+}) => (
+    <Coordinates.Cartesian
+        xAxis={{
+            lines: props.step[0],
+            subdivisions: props.step[0] / props.gridStep[0],
+            labels: (n) => (renderLineLabel(n, props.range[0]) ? n : ""),
```

Comment:
I _think_ in Graphie the axis labels are rendered with KaTeX, but I'd be curious to see if that's really ever needed (like do we only ever render numbers or can there be more complex expressions in the axis labels).  I suspect the answer is "numbers only".